[Doc Feature][Targeted messages Ph 1.5 - GA][5518210] (#14428)
MicrosoftDocs/msteams-docs: msteams-platform/agents-in-teams/agent-slash-commands.md

Reviewer: ```suggestion
### Slash and @mention commands for agents
```

This section should cover named commands for agents, including both triggers (slash and @mention), and the user experience and how they work for each.

Author: Updated

Reviewer: In terms of the amount of text needed to explain, this is better than the above. The text should say what the result of this example is (draftCommand and reviewCommand are made available as slash commands in both the compose and command boxes; summarizeCommand is not.)

Reviewer: ```suggestion
## Implementing agent named command handling
```

This also needs to include considerations for @mention-triggered commands.

Author: @niravkapoor @dampbrownshoes Please share any information that must be added for commands using @metion trigger.

Reviewer: Once this is reworked to be about _named_ commands, targeted messaging shouldn't be considered a "type of slash command" anymore, but it should get a very short section in this article mentioning how it relates to slash commands (similar to what's already there now below, but even shorter) with a link to its article.

> ## What is a slash command

Reviewer: ```suggestion
## User experience
```

Author: Updated

Message extension slash commands can be configured to appear as autocomplete entries in both the compose box and the Teams command box.

## Mention commands in agents

Use @mention commands in your agents and apps when users need to address a specific agent or bot directly. The message is delivered to the mentioned agent, and the agent responds according to the conversation context. For example, a user can @mention a project management agent in a channel and enter assign to create a follow-up task from the conversation context.

For more information about @mention commands, see [@Mention](/microsoftteams/platform/teams-sdk/essentials/sending-messages/overview?pivots=csharp).

## Add commands

Slash commands and [targeted messaging](#targeted-messaging) are configured in the app or agent [manifest](/microsoft-365/extensibility/schema/root-compose-extensions-commands?view=m365-app-prev&tabs=syntax&preserve-view=true). Declaring commands in the manifest makes commands visible in Teams, while the app or agent [implementation](#implementing-agent-named-command-handling) determines how each command is handled after the user sends it.

### Slash and @mention commands for agents

> Agent slash commands require an agent to opt in to receive targeted messages; see [Receive targeted messages](targeted-messages.md#receive-targeted-messages).

Reviewer: Do @mention commands?

          "commands": [
            {
              "title": "assign",
              "description": "Assign a follow-up task"
            }
          ]
        },
        {
          "scopes": ["personal", "team", "groupChat"],
          "triggers": ["slash", "mention"],
          "commands": [
            {
              "title": "review",
              "description": "Review a document"
            }
          ]
        }
      ]
    }
  ]
}
```

In this example:

- The agent is registered to operate in `personal`, `team`, and `groupChat` scopes. The
`supportsTargetedMessages` is set to `true` to allow the agent to receive targeted messages in group conversations.

- The `triggers` property defines where a command appears and how users can invoke it from the Teams compose experience. Use `slash` when the command should appear in the slash command picker, use `mention` when the command should appear in the traditional @mention command menu, or include both values when the same command should be available from both entry points.

  - The `summarize` command is configured to be a slash command.
  - The `assign` command appears as an @mention command option.

>   - The `review` command is available as both a slash command and an @mention command in the agent.

Reviewer: ```suggestion
- As is required for slash commands, `supportsTargetedMessages` is set to `true` to allow the agent to receive targeted messages in group conversations.
- `summarize` is configured as a slash command.
- `assign` is configured as an @mention command.
- `review` is configured as both a slash command an and @mention command.
```

Author: Updated

Invoking a command from the menu switches the compose box to targeted messaging mode for the agent and inserts the name of the command as the message text. Selecting **Send** sends the targeted message to the agent.

> Agents or apps can explicitly declare the commands your agent supports, and Teams shows them in the slash command picker when a user types `/`.  For example, a project management agent or bot app might expose commands such as `/create-task` to turn the current conversation into a tracked task based on context, or `/status-check` command that can be given a work item number and returns information about it. To support natural-language prompts, an agent must explicitly opt in. This keeps the behavior intentional.

Reviewer: Same comment as previously: remove this paragraph and work the examples into the above paragraphs. Remove "To support natural-language prompts, an agent must explicitly opt in. This keeps the behavior intentional."

Author: Following the earlier comment, I condensed this content from bullet points to make it more concise. I have now updated it as suggested.

Invoking a command from the menu switches the compose box to targeted messaging mode for the agent and inserts the name of the command as the message text. Selecting **Send** sends the targeted message to the agent.

> - **App-defined slash commands**:

Reviewer: The whole parent section of this, _Agent slash commands_, is already about "app-defined slash commands"; the examples here should be moved up there and compressed into a sentence, like: "For example, a project management agent might expose a command like `/create-task` that creates a new task item based on the conversation's context, or a `/status-check` command that can be given an item ID and returns information about it."

The "Natural-language support for slash commands" subsection should be removed - it is not a feature, what it is referring to is targeted messaging, which is already described above in a separate section. One concept you've got there - keeping the command set small and focused and generally emphasizing natural language instead - is a good candidate for the _Best practices and design guidance_ section at the bottom though.

Author: Updated

  - `/contoso find the latest rollout plan and highlight open risks`

  Natural-language support is often preferable to relying only on explicit commands for high-confidence, repeatable actions: agents are most valuable when they can understand intent, parameters, and context expressed in natural language, so slash-command design should combine discoverable commands with flexible phrasing that lets users say what they need instead of learning a long command catalog. From a platform perspective, slash commands should support the experience, not define it end to end; use them to surface common actions in a way that’s easy to find and invoke, then let the agent handle natural-language requests, infer missing details when appropriate, and ask brief follow-up questions when something is unclear. Keep the command set small so users don’t need to memorize syntax or choose between nearly identical options.

  To support natural-language prompts, an agent must explicitly opt in. This keeps the behavior intentional. Agents that are designed to interpret free-form requests can enable natural-language prompting and provide a more conversational experience.

> ### Message extension slash commands

Reviewer: ```suggestion
### Slash commands for action-type message extensions
```

Author: Updated

Agent slash commands require an agent to opt in to receive targeted messages; see [Receive targeted messages](targeted-messages.md#receive-targeted-messages).

#### Triggers: slash and @mention

> 

Reviewer: Show one representative example first, with a brief explanation of the behavior it results in, then describe the different aspects of it. If needed, a second example can go further down.

Author: @niravkapoor Please share a Teams SDK example for @mention.

Author: Updated

Use `slash` triggers for commands that users should be able to discover and invoke quickly from an empty compose box. Slash commands are well suited for frequent actions, private lookups, lightweight workflow starts, and commands that should be easy to find without first mentioning the agent.Use `@mention` triggers for commands that are closely tied to an explicit interaction with the agent. This pattern is useful when users are already addressing the agent directly, especially in personal scope or in conversations where the user expects the command to be part of a visible agent exchange.

- Use `@mention` when users are intentionally addressing the agent and may expect an agent-visible conversation flow.
- Use `slash` when users need fast discovery, private command invocation, or a shortcut from the compose box.
- Use both when the same command should be available through both discovery surfaces without changing the command behavior.

>     When a command supports both invocation patterns, declare both trigger values in the same command list. This keeps the command catalog consistent while giving users flexibility to start from either @mention or slash. You can choose to define separate command lists so that each trigger can have its own scope, command set, and user guidance.

Reviewer: These three paragraphs should be condensed to about two sentences, it's trying to include too much best practice and guidance. The definitions of these features, and concepts and information about best practices, is elsewhere in the article, this should focus on what the options are and what they do.

When a user activates an agent slash command, Teams switches the compose box to targeted message mode and inserts the command name, then returns control to the user to allow them to enter more text and send the message. The agent's message handler is responsible for using the contents of the message to determine whether it should be interpreted as a command invocation and handling it appropriately.

See [Best practices and design guidance](#best-practices-and-design-guidance) for other message handling guidance.

## Targeted messaging

Agents and users can interact using [targeted messages](targeted-messages.md) by using slash commands. In channels, group chats, and meeting chats, the slash command autocomplete menu shows the agents in the conversation that can receive targeted messages. Selecting an agent command switches the compose box to targeted message mode. A user-initiated targeted message is delivered only to the selected agent, and the agent can respond privately to the user.

For more information, see [send and receive targeted messages in group conversations](targeted-messages.md).

> 

Reviewer: ```suggestion
```

Reviewer: There is already a dedicated article for TM, including this much information about it here confuses things with the agent commands feature.

Author: Updated

> ## Best practices and design guidance

Reviewer: This should get a guidance item for choosing whether to surface a given command as slash or @mention.

Author: Updated
